app/controllers/news_controller.rb

```diff
@@ -0,0 +1,21 @@
+class NewsController < ApplicationController
+  before_action :set_page, only: :index
+
+  def show
+    @releases = Rubygem.joins(:latest_version)
+      .order("versions.created_at DESC")
+      .where("versions.created_at BETWEEN ? AND ?", Date.today - 1.year, Date.today)
+      .paginate(page: @page, per_page: 10, total_entries: 100)
+  end
+
+  def popular
+    @title = "New Releases — Popular Gems"
+    @releases = Rubygem.joins(:latest_version)
+      .by_downloads
+      .order("versions.created_at DESC")
+      .where("versions.created_at BETWEEN ? AND ?", Date.today - 1.year, Date.today)
+      .paginate(page: @page, per_page: 10, total_entries: 100)
```

Comment:
That said latest_version could be very old, so I don't want to rely on that. I want actually new version as defined by created between now and 7 days ago, for instance.